### PR TITLE
Add authorization

### PR DESCRIPTION
**Pull request by principles31**
# Blog app: add authorization rules:
- In this project, you will add authorization to your app using [CanCanCan](https://github.com/CanCanCommunity/cancancan).
- The following were implemented in this **PR as the project requirements using pair programming.**
> Installed CanCanCan as a gem .
> Added a role column to the users table
> A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.
- Implemented the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Used CanCanCan for this authorization.
- Implemented the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -24,6 +24,20 @@ def create
...
+    @comment.each(&:destroy)
+    @post.destroy
+    flash[:success] = ['Post Deleted Successfully']
+
+    respond_to do |format|
+      format.html { redirect_to "/users/#{current_user.id}/posts" }
+      format.json { head :no_content }
+    end
+  end
+
```
Comment (brytebee) [inline app/controllers/posts_controller.rb:40]: - [ ] **Required:** Using the `gem cancancan` as mentioned above, this file should be protected against unauthorized posts. Kindly consult the [cancancan documentation.](https://github.com/CanCanCommunity/cancancan#controller-helpers)

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @principles31 & @Kwentsir👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great schema design.✔️
- App is functional.✔️
- The project is professionally documented.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️

### I left this review here because the affected files haven't changed in this milestone.
- Your authorization should be handled with the `gem cancan` or `gem cancancan`, however, I cannot find it in your Gemfile. 

https://github.com/principles31/Blog_app/blob/cd7a31f1c1410f52a477ad26bd1be4ff542d0f9f/Gemfile#L1-L78

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
- [x] Required: Using the gem cancancan as mentioned above, this file should be protected against unauthorized posts. Kindly consult the [cancancan documentation.](https://github.com/CanCanCommunity/cancancan#controller-helpers)
- [x] The above requirement requested by the previous reviewer has been implemented.

**Comment by principles31** [comment]
@brytebee Thank you for your constructive feedback

**Review by sja-thedude — APPROVED** [review]
## Project Approved 👍 

Hi @principles31, This is @sja-thedude your code reviewer.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights:
- All linters passing
- All requirements completed
- Professional readme

giphy

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by principles31** [comment]
@sja-thedude Thanks for your approval.

**Pull request merged**